### PR TITLE
fix: harden registry schema validation and fixtures

### PR DESCRIPTION
**Pull request by TobyTheHutt**
## What  

Convert the governance registry schema to JSON Schema, add fixture-driven validation coverage, and document registry validation + container runtime overrides.

## Why  

We need stronger governance registry validation and repeatable fixtures for valid/invalid/edge cases, plus clearer docs for the schema/fixtures. Making the container runtime configurable keeps `make entity-model-erd` usable beyond Docker.

Resolves: #103 

## How  

- Replace `docs/schema/registry.schema.yaml` with `docs/schema/registry.schema.json` and wire schema loading/validation into `cmd/registry-check/main.go`.
- Add fixtures under `testutil/fixtures/registry/` with stub docs and exercise them via `cmd/registry-check/fixtures_test.go` plus additional parsing/validator tests.
- Document the registry schema + fixture layout in `docs/schema/README.md` and `testutil/fixtures/registry/README.md`.
- Add `DOCKER` override support in `Makefile` and document usage in `CONTRIBUTING.md` and `AGENTS.md`.

## Notes  

- Registry schema validation currently supports a focused JSON Schema subset; new keywords will need validator updates in `cmd/registry-check/main.go`.

## Checklist  

- [x] I have used conventional commits (conventionalcommits.com)
- [x] I have updated the documentation
- [ ] I have updated the relevant RFC and its linked resources
- [x] I have added tests
- [x] I ran manual tests
- [x] I strictly followed `AGENTS.md`

# Release Notes: Registry Schema Validation Hardening (PR #110)

## Overview
Migrates the governance registry schema from YAML to JSON Schema, integrates schema-driven validation into the registry-check CLI, adds fixture-driven tests and docs, and makes container runtime configurable in the Makefile. Resolves issue #103.

## What changed
- Schema
  - Replaced docs/schema/registry.schema.yaml with docs/schema/registry.schema.json.
  - Schema enforces top-level documents array (minItems:1) and per-document required fields: id, type, title, status, path. Constrains enums (type: RFC|Annex|ADR; status: Draft|Planned|Accepted|Superseded|Archived), date fields, quorum pattern, non-empty string arrays, and disallows additionalProperties.

- Validation & CLI
  - cmd/registry-check/main.go now loads the JSON Schema and validates docs/rfc/registry.yaml at runtime.
  - Introduces an internal jsonSchema representation and recursive validation helpers (format validators, pattern/min constraints, required/additionalProperties handling, clear error path reporting).
  - Document struct annotated with JSON tags to support schema-based checks and JSON conversion.

- Tests & Fixtures
  - New fixture tree testutil/fixtures/registry/ with valid, invalid, edge, and docs fixtures.
  - New tests (cmd/registry-check/fixtures_test.go and related test files) exercise schema loading, parsing, value rules, document-to-map conversion, repo-root discovery, and status handling.
  - Test harness improvements: TestMain/runTests with findRepoRoot to make tests resilient to working directory.

- Documentation & DX
  - Added docs/schema/README.md and testutil/fixtures/registry/README.md describing schema and fixture layout.
  - Updated CONTRIBUTING.md and AGENTS.md to document DOCKER override and guidance for container-backed tasks.
  - Makefile: added DOCKER variable (default docker) and runtime availability pre-check; entity-model-erd targets use $(DOCKER) for pluggable runtimes.

- Tooling & CI
  - .github/workflows: added R setup step for CI (r-lib/actions/setup-r@v2).
  - scripts/run_lintr.py: improved R env sanitization and bumped xml2 pin to 1.5.0.
  - .golangci.yml: allowed net/mail for email validation.
  - No public API surface changes; CLI remains entry point.

## Compliance assessment vs repo standards (CONTRIBUTING.md, ARCHITECTURE.md, pull_request_template)
- Strengths
  - Fixtures and tests: Meets and exceeds fixture-driven coverage requirements described in the PR objectives; tests cover happy-path, error cases, and edge conditions.
  - Documentation: Adds schema and fixture READMEs and updates contributing/agents docs for developer guidance.
  - CI & developer workflow: Makes registry validation available in tooling and documents local overrides (DOCKER), improving reproducible local runs.
  - Non-invasive: Changes are tooling/testing focused (cmd/registry-check, testutil, docs, Makefile) and do not alter public APIs or generated artifacts.

- Notable gap
  - RFC/ADR tracking: PR metadata and description state "Relevant RFC updated: no." Per repository standards (ARCHITECTURE.md and pull_request_template expectations), design-affecting changes should be captured in RFC/ADR form. The implementation and schema are provided without a corresponding RFC/ADR recording the design decision and rationale — this is a standards nonconformance to address (add an RFC/ADR that documents the JSON Schema choice, focused schema subset support, and extension rules).

- Scope limitation (documented)
  - Validator supports a focused subset of JSON Schema keywords; adding additional keywords will require updates in cmd/registry-check/main.go. This limitation is documented in PR notes; teams should be aware when evolving the schema.

## Risk & migration notes
- Backwards compatibility: No breaking changes to existing registry documents beyond stricter validation; consumers must ensure registry fixtures conform to the new JSON Schema (path patterns, enums, non-empty strings, and additionalProperties prohibition).
- Migration: Update any tooling or scripts that referenced the old YAML schema path to docs/schema/registry.schema.json. The CLI enforces the new schema via make targets (entity-model-verify / registry-lint).

## Tests & verification
- New unit tests added across cmd/registry-check; test harness updated to locate repo root for deterministic fixture resolution.
- Fixtures include minimal, multi, and full valid examples plus invalid/missing-field/path cases and edge-case empty lists/status-header parsing.

## Action items before merge (recommended)
- Add an RFC/ADR capturing the schema migration, rationale for the focused JSON Schema subset, and the process for extending supported keywords.
- Document expected validator extension process and add a short CONTRIBUTING note about adding JSON Schema keywords and corresponding validator updates.
- Optional: Consider extracting more JSON Schema features into a reusable validator library or adopting a maintained JSON Schema library if broader schema coverage is required.

## Conclusion
This PR delivers the core objectives: schema migration to JSON Schema, runtime validation wired into registry-check, comprehensive fixture-driven tests, and improved developer ergonomics for container runtimes and CI. Addressing the missing RFC/ADR will bring the change into full compliance with repository governance standards.

### CODE REVIEW

**Comment by TobyTheHutt** [comment]
@coderabbitai resolve
@coderabbitai summary
@coderabbitai full review

**Comment by TobyTheHutt** [comment]
@coderabbitai generate docstrings

**Comment by TobyTheHutt** [comment]
@coderabbitai full review

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -63,18 +63,19 @@ Use the issue forms:
   - Test: `make test`
   - Lint/format: `make lint` (runs gofmt/vet/registry/golangci plus Ruff and the R lintr)
 - Import guardrails rely on `import-boss`; the runbook in `docs/annex/0003-import-boss-runbook.md` covers command syntax and troubleshooting.
+- Container runtime: Makefile targets that spin up containers (for example `make entity-model-erd`) use `DOCKER` (default `docker`). Override with `DOCKER=podman` or `DOCKER=nerdctl` if needed, and ensure the runtime supports `docker`-compatible CLI behavior including `--network container:` semantics.
```
[the remark this answers is not in the document]
Comment (TobyTheHutt) [reply CONTRIBUTING.md]: Won't fix.

**Comment by TobyTheHutt** [comment]
@coderabbitai you mention multiple learnings in your earlier comments, but they are not stored in the account. Assert that your learnings get persisted.
